File: src/main/java/com/imbananko/tilly/utility/SqlQueries.java
package com.imbananko.tilly.utility;

import io.vavr.collection.HashMap;
import io.vavr.collection.Map;
import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "sql")
@Getter
public class SqlQueries {
  private Map<String, String> queries;

  public void setQueries(java.util.Map<String, String> queries) {
      this.queries = HashMap.ofAll(queries);
  }
}

File: src/main/kotlin/com/imbananko/tilly/utility/SqlQueries.kt
package com.imbananko.tilly.utility

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "sql")
data class SqlQueries(val queries: Map<String, String>) {
  fun getFromConfOrFail(key: String): String =
      this.queries[key] ?: error("Configuration should contain `$key`")
}
